### PR TITLE
QR code address

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] Show qr in receive dialog
- [x] Scan qr in send dialog
- [x] Open settings works

### CODE REVIEW

File: shared/app/nav.native.js [hunk 1]
```diff
@@ -352,7 +352,17 @@ class Nav extends Component<Props, {keyboardShowing: boolean}> {
       />
     )
     const layerScreens = this.props.routeStack.filter(r => r.tags && r.tags.layerOnTop)
-    const layers = layerScreens.map(r => r.leafComponent({shouldRender: true}))
+    const layers = layerScreens.map(
```
Comment (chrisnojima) [inline shared/app/nav.native.js]: these totally ignored the tags! Not adding more right now but likely this needs to change

File: shared/chat/routes.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ import ChooseEmoji from './conversation/messages/react-button/emoji-picker/conta
 import ConfirmForm from '../wallets/confirm-form/container'
 import SendForm from '../wallets/send-form/container'
 import ChooseAsset from '../wallets/send-form/choose-asset/container'
+import QRScan from '../wallets/qr-scan/container'
```
Comment (chrisnojima) [inline shared/chat/routes.js]: add scan and make all the form stuff be full screen like on wallets

File: shared/common-adapters/index.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ export {default as Banner} from './banner'
 export {Box, Box2} from './box'
 export {default as ButtonBar} from './button-bar'
 export {default as Button} from './button'
+export {default as QRLines} from './qr-lines'
```
Comment (chrisnojima) [inline shared/common-adapters/index.js]: this is used in multiple places now

File: shared/common-adapters/info-note.js [hunk 4]
```diff
@@ -7,15 +7,16 @@ import {isMobile} from '../constants/platform'
 
 export type Props = {
   containerStyle?: StylesCrossPlatform,
-  children: React.Node,
+  children?: React.Node,
+  color?: ?string,
```
Comment (chrisnojima) [inline shared/common-adapters/info-note.js]: let you pass in a color

File: shared/wallets/send-form/participants/search.js [hunk 5]
```diff
@@ -28,6 +30,7 @@ class Search extends React.Component<SearchProps, SearchState> {
   state = {
     displayResultsList: false,
     hideClearSearch: true,
+    searchText: '',
```
Comment (chrisnojima) [inline shared/wallets/send-form/participants/search.js]: we want to hide the scan if there's text

**Comment by chrisnojima** [comment]
@keybase/react-hackers this works now

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/app/nav.native.js (hunk 1 above)
Comment (buoyad) [reply shared/app/nav.native.js]: These mess up the send form on iPhone X, the back button on the main form and 'Cancel' on the qr scan are unreachable

File: shared/wallets/receive-modal/index.js [hunk 6]
```diff
@@ -38,66 +39,82 @@ const ReceiveModal = (props: Props) => {
...
+      <Kb.Text type="BodySmallSemibold">Your Stellar QR code:</Kb.Text>
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.qrContainer} centerChildren={true}>
+        <QrImage address={stellarAddress} />
+      </Kb.Box2>
+    </Kb.Box2>
+  </Kb.Box2>
 )
 
+const QrImage = ({address}) => {
+  const qr = QRCodeGen(4, 'L')
```
Comment (buoyad) [inline shared/wallets/receive-modal/index.js]: This looks very small on mobile. maybe we can up it a little?

File: shared/wallets/receive-modal/index.js [hunk 7]
```diff
@@ -38,66 +39,82 @@ const ReceiveModal = (props: Props) => {
...
-    </Box2>
-  </Box2>
+    <Kb.Box2 direction="vertical" gap="xtiny" fullWidth={true} style={styles.stellarAddressesContainer}>
+      <Kb.Text type="BodySmallSemibold">Your public Stellar address:</Kb.Text>
+      <Kb.CopyText buttonType="Wallet" text={stellarAddress} />
+    </Kb.Box2>
+    <Kb.Box2 direction="vertical" gap="xtiny" fullWidth={true} style={styles.stellarAddressesContainer}>
+      <Kb.Text type="BodySmallSemibold">Your Stellar QR code:</Kb.Text>
+      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.qrContainer} centerChildren={true}>
+        <QrImage address={stellarAddress} />
```
Comment (buoyad) [inline shared/wallets/receive-modal/index.js]: If this is the default account we should use the federated address here so this ends up as a direct payment

**Comment by chrisnojima** [comment]
- [ ] need iphone xr fix in first

**Comment by chrisnojima** [comment]
- [ ] switch to federated

File: shared/app/nav.native.js (hunk 1 above)
Comment (chrisnojima) [reply shared/app/nav.native.js]: iphone x stuff should be fixed in master and merged back into here

**Comment by chrisnojima** [comment]
@buoyad updated, ptal

File: shared/provision/code-page/qr-scan/index.native.js [hunk 8]
```diff
@@ -18,13 +16,13 @@ const QRScan = (props: Props) => (
         flashMode={RNCamera.Constants.FlashMode.off}
         permissionDialogTitle={'Permission to use camera'}
         permissionDialogMessage={'We need access to your camera to scan in the secret code'}
-        notAuthorizedView={<QRScanNotAuthorized onOpenSettings={props.onOpenSettings} />}
+        notAuthorizedView={<QRNotAuthorized onOpenSettings={props.onOpenSettings} />}
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/provision/code-page/qr-scan/index.native.js]: done

File: shared/wallets/qr-scan/index.native.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    </Kb.Text>
+    <Kb.Box2 direction="vertical" fullWidth={true} style={styles.topContainer} gap="small">
+      <Kb.Box2 direction="vertical" style={styles.cameraContainer}>
+        <RNCamera
+          type={RNCamera.Constants.Type.back}
+          autoFocus={RNCamera.Constants.AutoFocus.on}
+          captureAudio={false}
+          flashMode={RNCamera.Constants.FlashMode.off}
+          permissionDialogTitle={'Permission to use camera'}
+          permissionDialogMessage={'We need access to your camera to scan in the secret code'}
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/wallets/qr-scan/index.native.js]: done

File: shared/wallets/send-form/participants/to-field.js [hunk 10]
```diff
@@ -105,19 +108,31 @@ class ToStellarPublicKey extends React.Component<ToStellarPublicKeyProps, ToStel
...
+              onChangeText={this._onChangeRecipient}
+              textType="BodySemibold"
+              placeholder={'Stellar address\nEx: G12345... or you*example.com'}
+              placeholderColor={Styles.globalColors.black_20}
+              hideBorder={true}
+              containerStyle={styles.input}
+              multiline={true}
+              rowsMin={2}
+              rowsMax={3}
+              value={this.state.recipientPublicKey}
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/wallets/send-form/participants/to-field.js]: done

File: shared/wallets/send-form/participants/to-field.js [hunk 11]
```diff
@@ -105,19 +108,31 @@ class ToStellarPublicKey extends React.Component<ToStellarPublicKeyProps, ToStel
...
+            />
+          </Kb.Box2>
+          {!this.props.recipientPublicKey &&
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/wallets/send-form/participants/to-field.js]: done

File: shared/wallets/qr-scan/index.native.js [hunk 12]
```diff
@@ -0,0 +1,93 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import {RNCamera} from 'react-native-camera'
+import type {Props} from '.'
+
+const QRScan = (props: Props) => (
+  <Kb.Box2 direction="vertical" fullWidth={true} fullHeight={true} style={styles.container}>
+    <Kb.Text type="BodyBigLink" style={styles.cancel} onClick={() => props.onSubmitCode()}>
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/wallets/qr-scan/index.native.js]: put in a kinda crappy fix for this. we'll revisit when we redo the nav

File: shared/wallets/send-form/participants/to-field.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/wallets/send-form/participants/to-field.js]: hm, it's still `props`

**Pull request merged**